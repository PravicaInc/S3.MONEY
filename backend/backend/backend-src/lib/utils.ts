/**
 * Convert smart contract ticker to package name.
 *
 * @param {string} ticker - smart contract ticker, e.g. $SUI
 * @returns {string} - package name, e.g. sui
 */
export function tickerToPackageName(ticker: string): string {
  return ticker.toLowerCase().trim()
    .slice(1);
}

// eof
