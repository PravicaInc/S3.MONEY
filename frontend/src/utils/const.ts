export const PAGES_URLS = {
  home: '/',
  dashboardOverview: '/dashboard/overview',
  dashboardOperations: '/dashboard/operations',
  signIn: '/sign-in',
  createStableCoin: '/create-stablecoin',
};

export const WALLETS = {
  SuiWallet: 'Sui Wallet',
};
