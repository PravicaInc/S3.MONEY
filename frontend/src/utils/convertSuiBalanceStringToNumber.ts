const convertSuiBalanceStringToNumber = (balance: string): number => parseInt(balance) / 10e8;

export default convertSuiBalanceStringToNumber;
