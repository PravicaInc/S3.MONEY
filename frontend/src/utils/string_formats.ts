export const numberNormalize = (value: string = ''): number => Number(`${value}`.replace(/\D+/g, ''));

export const numberFormat = (value: string): string => value
  ? new Intl.NumberFormat('en-US').format(numberNormalize(value))
  : value;

export const priceFormat = (value: string): string => value
  ? new Intl.NumberFormat('en-US', {
    style: 'currency',
    currency: 'USD',
    maximumFractionDigits: 2,
    minimumFractionDigits: value.indexOf('.') !== -1 ? 2 : 0,
  })
    .format(numberNormalize(value))
  : value;

export const getShortAccountAddress = (accountAddress: string, partLength: number = 4) => (
  `${accountAddress.substring(0, partLength)}...${accountAddress.substring(accountAddress.length - partLength)}`
);

export function shortNumberFormat(num: number, precision = 2): string {
  const map = [
    { suffix: 'T', threshold: 1e12 },
    { suffix: 'B', threshold: 1e9 },
    { suffix: 'M', threshold: 1e6 },
    { suffix: 'K', threshold: 1e3 },
    { suffix: '', threshold: 1 },
  ];

  const found = map.find(x => Math.abs(num) >= x.threshold);

  return found
    ? (num / found.threshold).toFixed(precision) + found.suffix
    : `${num}`;
}
