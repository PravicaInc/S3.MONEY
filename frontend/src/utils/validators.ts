export const suiAddressRegExp = /^0[xX][a-fA-F0-9]{64}$/;

export const isSuiAddress = (address: string) => suiAddressRegExp.test(address);
