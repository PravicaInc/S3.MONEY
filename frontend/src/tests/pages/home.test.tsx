import { useAutoConnectWallet, useCurrentAccount } from '@mysten/dapp-kit';
import { cleanup, RenderResult } from '@testing-library/react';

import '@testing-library/jest-dom';

import HomePage from '@/app/home/page';

import { createMockAccount } from '@/tests/utils/create_mock_wallet_account';
import { renderWithProviders } from '@/tests/utils/render_with_providers';

const renderHomePageWithProviders = () => renderWithProviders(<HomePage />);

let renderResult: RenderResult;

describe('Home page:', () => {
  afterAll(cleanup);

  describe('Home page with loading state:', () => {
    beforeEach(() => {
      if (renderResult?.unmount) {
        renderResult.unmount();
      }

      jest.mocked(useAutoConnectWallet).mockImplementation(() => 'idle');

      renderResult = renderHomePageWithProviders();
    });

    it('Show loader', () => {
      expect(renderResult.queryByTestId('loader')).toBeVisible();
    });
  });

  describe('Home page with disconnected state:', () => {
    beforeEach(() => {
      if (renderResult?.unmount) {
        renderResult.unmount();
      }

      jest.mocked(useAutoConnectWallet).mockImplementation(() => 'attempted');

      renderResult = renderHomePageWithProviders();
    });

    it('Show loader', () => {
      expect(renderResult.queryByTestId('loader')).toBeVisible();
    });
  });

  describe('Home page with connected state:', () => {
    beforeEach(() => {
      if (renderResult?.unmount) {
        renderResult.unmount();
      }

      jest.mocked(useAutoConnectWallet).mockImplementation(() => 'attempted');
      jest.mocked(useCurrentAccount).mockImplementation(createMockAccount);

      renderResult = renderHomePageWithProviders();
    });

    it('Don`t show loader', () => {
      expect(renderResult.queryByTestId('loader')).toBeNull();
    });

    it('Show stablecoin form', () => {
      expect(renderResult.queryByText('Select Stablecoin')).toBeVisible();
    });

    it('Snapshot', () => {
      expect(renderResult.container).toMatchSnapshot();
    });
  });
});
