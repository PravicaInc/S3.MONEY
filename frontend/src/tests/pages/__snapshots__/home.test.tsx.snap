// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home page: Home page with connected state: Snapshot 1`] = `
<div>
  <style>
    [data-dapp-kit] {--dapp-kit-blurs-modalOverlay:blur(0);--dapp-kit-backgroundColors-primaryButton:#F6F7F9;--dapp-kit-backgroundColors-primaryButtonHover:#F0F2F5;--dapp-kit-backgroundColors-outlineButtonHover:#F4F4F5;--dapp-kit-backgroundColors-modalOverlay:rgba(24 36 53 / 20%);--dapp-kit-backgroundColors-modalPrimary:white;--dapp-kit-backgroundColors-modalSecondary:#F7F8F8;--dapp-kit-backgroundColors-iconButton:transparent;--dapp-kit-backgroundColors-iconButtonHover:#F0F1F2;--dapp-kit-backgroundColors-dropdownMenu:#FFFFFF;--dapp-kit-backgroundColors-dropdownMenuSeparator:#F3F6F8;--dapp-kit-backgroundColors-walletItemSelected:white;--dapp-kit-backgroundColors-walletItemHover:#3C424226;--dapp-kit-borderColors-outlineButton:#E4E4E7;--dapp-kit-colors-primaryButton:#373737;--dapp-kit-colors-outlineButton:#373737;--dapp-kit-colors-iconButton:#000000;--dapp-kit-colors-body:#182435;--dapp-kit-colors-bodyMuted:#767A81;--dapp-kit-colors-bodyDanger:#FF794B;--dapp-kit-radii-small:6px;--dapp-kit-radii-medium:8px;--dapp-kit-radii-large:12px;--dapp-kit-radii-xlarge:16px;--dapp-kit-shadows-primaryButton:0px 4px 12px rgba(0, 0, 0, 0.1);--dapp-kit-shadows-walletItemSelected:0px 2px 6px rgba(0, 0, 0, 0.05);--dapp-kit-fontWeights-normal:400;--dapp-kit-fontWeights-medium:500;--dapp-kit-fontWeights-bold:600;--dapp-kit-fontSizes-small:14px;--dapp-kit-fontSizes-medium:16px;--dapp-kit-fontSizes-large:18px;--dapp-kit-fontSizes-xlarge:20px;--dapp-kit-typography-fontFamily:ui-sans-serif, system-ui, -apple-system, BlinkMacSystemFont, "Segoe UI", Roboto, "Helvetica Neue", Arial, "Noto Sans", sans-serif, "Apple Color Emoji", "Segoe UI Emoji", "Segoe UI Symbol", "Noto Color Emoji";--dapp-kit-typography-fontStyle:normal;--dapp-kit-typography-lineHeight:1.3;--dapp-kit-typography-letterSpacing:1;}
  </style>
  <div
    class="flex items-center gap-2 px-6 min-h-20 border-b border-borderPrimary bg-white justify-between"
  >
    <div
      class="relative"
    >
      <input
        class="text-primary placeholder:text-hitGrey p-3 rounded-xl border border-borderPrimary focus:border-actionPrimary outline-none disabled:text-[#787878] disabled:bg-white pl-11 w-[460px] text-sm py-[9px]"
        placeholder="Search..."
        type="text"
        value=""
      />
      <svg
        class="absolute top-1/2 left-3 -translate-y-1/2"
      />
    </div>
    <button
      class="border border-borderPrimary rounded-[10px] h-10 px-4 flex items-center justify-between gap-2 text-secondary font-medium text-sm transition shadow-logoutButton hover:bg-actionPrimary hover:bg-opacity-5 hover:border-actionSecondary"
    >
      0x11...1111
      <svg />
    </button>
  </div>
  <div
    class="overflow-auto h-full flex flex-col justify-between"
  >
    <div
      class="grid grid-cols-1 lg:grid-cols-2 xl:grid-cols-3 m-8 gap-6"
    >
      <div
        class="border border-borderPrimary p-4 bg-white rounded-2xl flex flex-col justify-between"
      >
        <div>
          <p
            class="text-primary font-medium flex items-center justify-between"
          >
            Contract Transaction
            <a
              class="text-secondary underline"
              href="https://suiscan.xyz/testnet/tx/1"
              rel="noreferrer"
              target="_blank"
            >
              1
            </a>
          </p>
          <div
            class="border border-borderPrimary rounded-lg overflow-hidden mt-4"
          >
            <div
              class="flex items-center gap-2 py-2 px-4 bg-whiteLilac border-b border-borderPrimary relative z-10"
            >
              <div
                class="w-7 h-7 rounded-full bg-seashell bg-no-repeat bg-center bg-cover shrink-0"
                style="background-image: url(undefined);"
              />
              <p
                class="flex items-center gap-1 text-ebonyClay text-sm font-medium"
              >
                <span
                  class=""
                >
                  Pravica Token
                </span>
                <span
                  class=""
                >
                  (
                  PRV
                  )
                </span>
              </p>
            </div>
            <div
              class="h-[176px] flex items-center justify-center"
            >
              <svg
                aria-hidden="true"
                class="svg-inline--fa fa-spinner animate-spin text-actionPrimary h-10"
                data-icon="spinner"
                data-prefix="fas"
                data-testid="loader"
                focusable="false"
                role="img"
                viewBox="0 0 512 512"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M304 48a48 48 0 1 0 -96 0 48 48 0 1 0 96 0zm0 416a48 48 0 1 0 -96 0 48 48 0 1 0 96 0zM48 304a48 48 0 1 0 0-96 48 48 0 1 0 0 96zm464-48a48 48 0 1 0 -96 0 48 48 0 1 0 96 0zM142.9 437A48 48 0 1 0 75 369.1 48 48 0 1 0 142.9 437zm0-294.2A48 48 0 1 0 75 75a48 48 0 1 0 67.9 67.9zM369.1 437A48 48 0 1 0 437 369.1 48 48 0 1 0 369.1 437z"
                  fill="currentColor"
                />
              </svg>
            </div>
          </div>
        </div>
        <a
          class="rounded-xl block"
          href="/dashboard/operations?txid=1"
        >
          <button
            class="bg-white text-primary rounded-xl shadow-button border-borderPrimary border-solid border font-semibold flex items-center justify-center transition hover:border-actionSecondary hover:text-actionPrimary disabled:bg-alabaster disabled:bg-none disabled:border-borderPrimary disabled:text-santaGrey disabled:shadow-none mt-4 w-full h-[48px]"
          >
            Manage
          </button>
        </a>
      </div>
      <div
        class="border border-borderPrimary p-4 bg-white rounded-2xl flex flex-col justify-between"
      >
        <div>
          <p
            class="text-primary font-medium flex items-center justify-between"
          >
            Contract Transaction
            <a
              class="text-secondary underline"
              href="https://suiscan.xyz/testnet/tx/2"
              rel="noreferrer"
              target="_blank"
            >
              2
            </a>
          </p>
          <div
            class="border border-borderPrimary rounded-lg overflow-hidden mt-4"
          >
            <div
              class="flex items-center gap-2 py-2 px-4 bg-whiteLilac border-b border-borderPrimary relative z-10"
            >
              <div
                class="w-7 h-7 rounded-full bg-seashell bg-no-repeat bg-center bg-cover shrink-0"
                style="background-image: url(undefined);"
              />
              <p
                class="flex items-center gap-1 text-ebonyClay text-sm font-medium"
              >
                <span
                  class=""
                >
                  S3 Money Token
                </span>
                <span
                  class=""
                >
                  (
                  SSS
                  )
                </span>
              </p>
            </div>
            <div
              class="h-[176px] flex items-center justify-center"
            >
              <svg
                aria-hidden="true"
                class="svg-inline--fa fa-spinner animate-spin text-actionPrimary h-10"
                data-icon="spinner"
                data-prefix="fas"
                data-testid="loader"
                focusable="false"
                role="img"
                viewBox="0 0 512 512"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M304 48a48 48 0 1 0 -96 0 48 48 0 1 0 96 0zm0 416a48 48 0 1 0 -96 0 48 48 0 1 0 96 0zM48 304a48 48 0 1 0 0-96 48 48 0 1 0 0 96zm464-48a48 48 0 1 0 -96 0 48 48 0 1 0 96 0zM142.9 437A48 48 0 1 0 75 369.1 48 48 0 1 0 142.9 437zm0-294.2A48 48 0 1 0 75 75a48 48 0 1 0 67.9 67.9zM369.1 437A48 48 0 1 0 437 369.1 48 48 0 1 0 369.1 437z"
                  fill="currentColor"
                />
              </svg>
            </div>
          </div>
        </div>
        <a
          class="rounded-xl block"
          href="/dashboard/operations?txid=2"
        >
          <button
            class="bg-white text-primary rounded-xl shadow-button border-borderPrimary border-solid border font-semibold flex items-center justify-center transition hover:border-actionSecondary hover:text-actionPrimary disabled:bg-alabaster disabled:bg-none disabled:border-borderPrimary disabled:text-santaGrey disabled:shadow-none mt-4 w-full h-[48px]"
          >
            Manage
          </button>
        </a>
      </div>
    </div>
    <div
      class="font-medium flex items-center justify-between text-mistBlue pb-7 pl-9 pr-[85px]"
    >
      <a
        href="https://pravica.io/privacy"
        rel="noreferrer"
        target="_blank"
      >
        Privacy Policy
      </a>
      <p>
        Copyright
         
        2024
      </p>
    </div>
  </div>
</div>
`;
